src/app/options/message/message.component.html

```diff
@@ -1,45 +1,105 @@
-<h2>{{'options.message.customMsg' | translate}}</h2>
+<h2>{{ "options.message.customMsg" | translate }}</h2>
 <form #messageOptions="ngForm" novalidate>
   <div class="input">
-    <label>{{'options.message.msg' | translate}}</label>
-    <options-toggle name="message" id="message" [(ngModel)]="settings.config.message.enabled" (ngModelChange)="shared.toggleOrder(settings.config.message.id, $event); ga.field('message.enabled', settings.config.message.enabled);"></options-toggle>
+    <label>{{ "options.message.msg" | translate }}</label>
+    <options-toggle
+      name="message"
+      id="message"
+      [(ngModel)]="settings.config.message.enabled"
+      (ngModelChange)="
+        shared.toggleOrder(settings.config.message.id, $event);
+        ga.field('message.enabled', settings.config.message.enabled)
+      "
+    ></options-toggle>
   </div>
   <ng-container *ngIf="settings.config.message.enabled">
     <div class="input">
-      <label>{{'options.message.msgText' | translate}}</label>
-      <input type="text" name="messageText" id="messageText" [(ngModel)]="settings.config.message.text" (ngModelChange)="ga.field('message.text', 'userData');" autocomplete="off">
+      <label>{{ "options.message.msgText" | translate }}</label>
+      <input
+        type="text"
+        #textMessage
+        name="messageText"
+        id="messageText"
+        [(ngModel)]="settings.config.message.text"
+        (ngModelChange)="ga.field('message.text', 'userData')"
+        autocomplete="off"
+      />
+      <div class="btn" (click)="addMessage()">Add</div>
```

Comment:
Given my suggestion elsewhere, there shouldn't be the old messages variable anymore so will need some tweaking. I see right now you do not have any way to edit previously added message. I think it would be best to set this up like the Quick Links where each quick link can be edited any time and a new one can always be added at the end. Please try to replicate the way that works here.

Reply:
fixed to work the same as QuickLinks Commit 50365ee0df7cb83591d874b21bbbe07063548203